apps/opencs/view/render/instancemode.cpp

```diff
@@ -681,3 +697,183 @@ void CSVRender::InstanceMode::deleteSelectedInstances(bool active)
 
     getWorldspaceWidget().clearSelection (SceneUtil::Mask_EditorReference);
 }
+
+void CSVRender::InstanceMode::dropInstance(DropMode dropMode, CSVRender::Object* object, float objectHeight)
+{
+    const osg::Vec3d& point = object->getPosition().asVec3();
+
+    osg::Vec3d start = point;
+    start.z() += objectHeight;
+    osg::Vec3d end = point;
+    end.z() = std::numeric_limits<float>::lowest();
+
+    osg::ref_ptr<osgUtil::LineSegmentIntersector> intersector (new osgUtil::LineSegmentIntersector(
+        osgUtil::Intersector::MODEL, start, end) );
+    intersector->setIntersectionLimit(osgUtil::LineSegmentIntersector::NO_LIMIT);
+    osgUtil::IntersectionVisitor visitor(intersector);
+
+    if (dropMode == Terrain_sep) visitor.setTraversalMask(SceneUtil::Mask_Terrain);
```

Comment:
We don't really do this kind of one-liners for if-clauses in OpenMW. The body of the if statement belongs into the next line. I have seen this a bunch of times.